lib/controller.dart

```diff
@@ -38,19 +59,58 @@ void init() async {
 }
 
 void onLoginCompleted() async {
+  view.showLoading();
+
   await loadDataFromFirebase();
+  _filteredInteractions = _allInteractions;
+  _filteredComparisonInteractions = _allInteractions;
+
+  computeUniqFieldValues();
+  _selectedTab = _config.tabs.first.id;
+
   view.setNavlinkSelected(_currentNavLink);
   _navLinks['analyse'].render();
+
+  view.hideLoading();
 }
 
 void onLogoutCompleted() async {
   logger.debug('Delete all local data');
 }
 
 void loadDataFromFirebase() async {
-  view.showLoading();
-  logger.debug('Read all required data');
-  view.hideLoading();
+  try {
+    _config = await fb.fetchConfig();
+  } catch (e) {
+    view.showAlert(_errorMessages['parse-config']);
+    logger.error(e);
+    rethrow;
+  }
+
+  try {
+    _allInteractions =
+        await fb.fetchInteractions(_config.data_paths['interactions']);
+  } catch (e) {
+    view.showAlert(_errorMessages['fetch-interactions']);
+    logger.error(e);
+    rethrow;
+  }
+}
+
+void computeUniqFieldValues() {
```

Comment:
a slightly better practice would be for this function to take the input data it needs as arguments, and then return its results. That way it's more modular, and it can get called multiple times if needs be. And then it'll be the job of the callee to do what it needs with the result (e.g. assign it to the global variable `_uniqueFieldValues`)

Reply:
Yes sure, i think it is also very explicit in what it is trying to do. Updated the function to take params, return the result than assign in the global variable